src/Sidekick/Windows/AdvancedSearch/AdvancedSearchView.xaml.cs

```diff
@@ -0,0 +1,289 @@
+            CurrentItem = item;
+            int rowCounter = 0;
+            RowElementsDictionary = new Dictionary<int, (TextBlock attrName, TextBox minVal, TextBox maxVal, CheckBox isChecked)>();
+
+            // Item Name and Type
+            gridAdvancedSearch.RowDefinitions.Add(BuildRow());
+            var itemNameBlock = BuildTextBlock(item.Name);
+            var itemTypeBlock = BuildTextBlock(item.Type);
+            itemNameBlock.Foreground = (SolidColorBrush)new BrushConverter().ConvertFrom(item.GetColor());
+            itemTypeBlock.Margin = new Thickness(150, 0, 0, 0);
+            gridAdvancedSearch.Children.Add(itemNameBlock);
+            gridAdvancedSearch.Children.Add(itemTypeBlock);
+            Grid.SetRow(itemNameBlock, rowCounter);
+            Grid.SetColumn(itemNameBlock, 0);
+            Grid.SetRow(itemTypeBlock, rowCounter);
+            Grid.SetColumn(itemTypeBlock, 1);
+            rowCounter++;
+
+            // Item Attributes
+            foreach (var pair in attributeItem.AttributeDictionary)
+            {
+                gridAdvancedSearch.RowDefinitions.Add(BuildRow());
+                var attributeBlock = BuildTextBlock(pair.Key.Text);
+                var minBlock = BuildTextBox(pair.Value.Min == null ? "" : pair.Value.Min.ToString());
+                var maxBlock = BuildTextBox(pair.Value.Max == null ? "" : pair.Value.Max.ToString());
+                var enabledCheckBox = new CheckBox() { HorizontalAlignment = HorizontalAlignment.Center, VerticalAlignment = VerticalAlignment.Center, IsChecked = true };
+                enabledCheckBox.Checked += EnableTextBoxes;
+                enabledCheckBox.Unchecked += DisableTextBoxes;
+                Grid.SetColumn(attributeBlock, AttributeColumnIndex);
+                Grid.SetRow(attributeBlock, rowCounter);
+                Grid.SetColumn(minBlock, MinValueColumnIndex);
+                Grid.SetRow(minBlock, rowCounter);
+                Grid.SetColumn(maxBlock, MaxValueColumnIndex);
+                Grid.SetRow(maxBlock, rowCounter);
+                Grid.SetColumn(enabledCheckBox, EnabledColumnIndex);
+                Grid.SetRow(enabledCheckBox, rowCounter);
+                gridAdvancedSearch.Children.Add(attributeBlock);
+                gridAdvancedSearch.Children.Add(minBlock);
+                gridAdvancedSearch.Children.Add(maxBlock);
+                gridAdvancedSearch.Children.Add(enabledCheckBox);
+                RowElementsDictionary.Add(rowCounter, (attributeBlock, minBlock, maxBlock, enabledCheckBox));
+                rowCounter++;
+            }
+
+            // Links + Sockets + Item Level + Influence
+            if ((item.GetType() == typeof(EquippableItem)))
+            {
+                gridAdvancedSearch.RowDefinitions.Add(BuildRow());
+                var equippableItem = (EquippableItem)item;
+                var socketCheckbox = new CheckBox() { Content = "Max Sockets" };        // TODO Best way to find max possible Sockets
```

Comment:
@Zalhera find the max socket/links the item has or max socket/links the item can have?

Reply:
I'm trying to find the max Numer of Sockets/Links the item can have.
(I'm mostly trying to replicate Poe Trade Macros Advanced Search Window for now)